client/js/screens/SignIn/SignIn.js

```diff
@@ -1,12 +1,99 @@
 import React from "react";
-import { View, Text, Button } from "react-native";
+import { View, Image, TouchableOpacity, TextInput, Text } from "react-native";
 import { styles } from "./styles";
+import PropTypes from "prop-types";
+import { Form, Field } from "react-final-form";
+
+const validate = values => {
+  const errors = {};
+  if (!values.email) {
+    errors.email = "Required";
+  }
+  if (!values.password) {
+    errors.password = "Required";
+  }
+  return errors;
+};
+
 const SignIn = ({ navigation }) => {
   return (
-    <View style={styles.SignInContainer}>
-      <Button title="Sign In" onPress={() => navigation.navigate("Layout")} />
+    <View style={styles.root}>
```

Comment:
Use KeyboardAvoidingView

Reply:
Wrap the whole design in KeyboardAvoidingView instead of doing it just for fields else your design will be destroyed once re-rendered.